featuretools/tests/primitive_tests/test_transform_features.py

```diff
@@ -1113,3 +1114,35 @@ def test_time_since_primitive_matches_all_datetime_types(es):
 
     for name in expected_names:
         assert name in fm.columns
+
+
+def test_cfm_with_lag_and_non_nullable_column(pd_es):
+    # fill nans so we can use non nullable numeric logical type in the EntitySet
+    new_log = pd_es['log'].copy()
+    new_log['value'] = new_log['value'].fillna(0)
+    new_log.ww.init(logical_types={'value': 'Integer',
+                                   "product_id": "Categorical"},
+                    index='id', time_index='datetime',
+                    name='new_log')
+    pd_es.add_dataframe(new_log)
+    rels = [('sessions', 'id', 'new_log', 'session_id'),
+            ('products', 'id', 'new_log', 'product_id')]
+    pd_es = pd_es.add_relationships(rels)
+
+    assert isinstance(pd_es['new_log'].ww.logical_types['value'], Integer)
+
+    lag_primitive = NumericLag(periods=5)
```

Comment:
Not a big deal, but you could use a variable for periods here and below instead of hardcoding `5` everywhere. That way it might be more obvious that the assertions are checking that the number of nulls introduced is equal to the periods used for lagging.

Reply:
done!